pybedlite/overlap_detector.py

```diff
@@ -126,36 +177,94 @@ def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
         )
 
 
-class OverlapDetector(Iterable[Interval]):
+_GenericGenomicSpan = TypeVar(
+    "_GenericGenomicSpan",
+    bound=Union[_GenomicSpanWithChrom, _GenomicSpanWithContig, _GenomicSpanWithRefName],
+)
+"""
+A generic genomic feature where the reference sequence name is accessed with any of the 3 most
+common property names ("chrom", "contig", "refname"). This type variable is used for describing the
+generic type contained within the :class:`~pybedlite.overlap_detector.OverlapDetector`.
+"""
+
+
+class OverlapDetector(Generic[_GenericGenomicSpan], Iterable[_GenericGenomicSpan]):
     """Detects and returns overlaps between a set of genomic regions and another genomic region.
 
-    Since :class:`~pybedlite.overlap_detector.Interval` objects are used both to populate the
-    overlap detector and to query it, the coordinate system in use is also 0-based open-ended.
+    The overlap detector may contain any interval-like Python objects that have the following
+    properties:
+
+      * `chrom` or `contig` or `refname`: The reference sequence name
+      * `start`: A 0-based start position
+      * `end`: A 0-based exclusive end position
+
+    Interval-like Python objects may also contain strandedness information which will be used
+    for sorting them in :func:`~pybedlite.overlap_detector.OverlapDetector.get_overlaps` using
+    either of the following properties if they are present:
+
+      * `negative (bool)`: Whether or not the feature is negative stranded or not
+      * `strand (BedStrand)`: The BED strand of the feature
+      * `strand (str)`: The strand of the feature (`"-"` for negative)
 
     The same interval may be added multiple times, but only a single instance will be returned
-    when querying for overlaps.  Intervals with the same coordinates but different names are
-    treated as different intervals.
+    when querying for overlaps.
 
     This detector is the most efficient when all intervals are added ahead of time.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, intervals: Optional[Iterable[_GenericGenomicSpan]] = None) -> None:
```

Comment:
what about passing as `*args` instead?

Reply:
Could do that too! My gut feeling is users have a large collection of intervals to put into an overlap detector in the default case, so a list already exists when commonly building this class. So, instead of splatting the list into a varargs, one could use that list directly. I think it is less common that users will have just a few intervals not in any collection to be input into the OverlapDetector.